obspy/io/nordic/core.py

```diff
@@ -723,34 +732,23 @@ def _read_picks(tagged_lines, new_event):
             polarity = line[16]
             if weight == ' ':
                 weight = 0
-        polarity_maps = {"": "undecidable", "C": "positive", "D": "negative"}
-        try:
-            polarity = polarity_maps[polarity]
-        except KeyError:
-            polarity = "undecidable"
+        polarity = POLARITY_MAPPING.get(polarity, "undecidable")
         # It is valid nordic for the origin to be hour 23 and picks to be hour
         # 00 or 24: this signifies a pick over a day boundary.
-        if int(line[18:20]) == 0 and evtime.hour == 23:
+        pick_hour = int(line[18:20])
+        pick_minute = int(line[20:22])
+        pick_seconds = float(line[22:29])  # 29 should be blank, but sometimes
+        # SEISAN appears to overflow here, see #2348
+        if pick_hour == 0 and evtime.hour == 23:
```

Comment:
Couldn't you get rid of lines 742 to 748 by just using strict == False in `UTCDateTime`, or was there some reason for not doing that?

Reply:
742-748 copes with stepping over a day boundary without any change in the day number.  It seems to be valid (I have had files that SEISAN made) that have an origin on one day (say 2010/1/1T23:59:20) and picks on the next day.  Picks in NORDIC do not contain the date, just hour, minutes and decimal seconds, so a pick could be formatted with hour=00 - this if statement should cope with that and know to add a day.
I could probably use `strict=False` to remove the `if pick_hour>=24` statement though, but I figured because I was doing this anyway it could just stay?

Reply:
Ah good point. I forgot the pick didn't include day info. No, this is certainly fine the way it is I just didn't understand at first.